SwaggerGenerator

```diff
@@ -174,7 +174,6 @@
             "type": "string",
             "name": "id",
             "in": "query",
-            "required": true,
```

Comment:
@rynowak, what's the best way to infer this? https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelMetadata.cs#L250 says it's only valid for properties and not parameters.

Reply:
The question is: When is a missing query parameter accepted or when does ASP.NET Core yield a validation error or throw an exception?

Reply:
ASP.Core doesn't do this by default anywhere (until 2.1 comes out). It has MVC's rules, all parameters are optional

Reply:
Could we surface that as a property on `ApiParameterDescription`? We can light it up if it's available.

Reply:
Sure. I think we might as well add it. Then NSwag can respect it and we can start setting it for things 👍 Please send a PR to MVC

Reply:
What about backward compatibility with WebApiToSwaggerGenerator? Do we accept the difference or should we fall back to the previous logic if the new property is missing?

Reply:
We need to have an eye to what you've done in the past (even if it's a lie). MVC will still call your action if there's no value, but the user might do something to enforce its validity. Even though MVC doesn't have a concept of 'required' - it may still be desirable from a documentation standpoint.

We're working on patching gaps like this in the runtime, but they have to be opt-in for new code.

Reply:
Part of the challenge for us is that MVC has some non-optimal and non-ideal behaviors for APIs. We're trying to update these as well as update our assumptions (in ApiExplorer) to support reasonable usage of the framework instead of all possible scenarios.

Reply:
Ok. We will see later... I think we should at least add a new attribute SwaggerRequiredAttribute (or use RequiredAttribute) to force a parameter to be required in the Swagger spec.

Reply:
Btw: Path parameters are always required, ref: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#fixed-fields-7